openmdao/core/problem.py

```diff
@@ -409,6 +409,16 @@ def setup(self, check=True, out_stream=sys.stdout):
         # a single source.
         connections = self._setup_connections(params_dict, unknowns_dict)
 
+        # Allow the user to omit the size of a parameter and pull the size
+        # and shape from the connection source.
+        for tgt, src in iteritems(connections):
+            tmeta = params_dict[tgt]
+            if not tmeta.get('pass_by_obj') and tmeta['shape'] == ():
+                smeta = unknowns_dict[src[0]]
+                tmeta['shape'] = smeta['shape']
+                tmeta['size'] = smeta['size']
+                tmeta['val'] = smeta['val']
```

Comment:
right.. you are implicitly setting the size/shape by specifying the indices.. maybe you don't feel the need to state them explicitly as well?  Maybe there is already code somewhere in the process that sets the size/shape based on src_indices, I'm not sure...  would be a good test case anyway, regardless of how we want to handle it, right?